src/arkreactor/Compiler/Parser.cpp

```diff
@@ -185,6 +195,13 @@ namespace Ark
         else if ((token.type == TokenType::Operator || token.type == TokenType::Identifier) &&
                  std::find(Builtins::operators.begin(), Builtins::operators.end(), token.token) != Builtins::operators.end())
             throwParseError("Found a free flying operator, which isn't authorized. Operators should always immediatly follow a `('.", token);
+        else if ((token.type == TokenType::Number ||
+                  token.type == TokenType::String) &&
+                 tokens.front().type == TokenType::Keyword)
+            throwParseError("Unexpected keyword `" + tokens.front().token + "' in the middle of an expression", token);
```

Comment:
The last argument here should be `tokens.front()` instead of `token`, so that the error context is related to the unexpected keyword and not what comes before it.